From: Arsen ArsenoviÄ‡ <arsen@gentoo.org>
Date: 1675754012 +0000
Subject: Bug 1811714 - Add a few missing <cstdint> includes r=gfx-reviewers,nical

GCC 13s libstdc++ reduced its dependency on some headers like <cstdint>,
so it's no longer transitively included through various headers.
Include it explicitly.

See also: https://gcc.gnu.org/gcc-13/porting_to.html#header-dep-changes

  11:16.32 .../components/telemetry/pingsender/pingsender.cpp:30:7:
                    error: unknown type name 'uint32_t'
  11:16.33 const uint32_t kConnectionTimeoutMs = 30 * 1000;
  11:16.33       ^
  11:16.33 ... /components/telemetry/pingsender/pingsender.cpp:76:9:
                    error: unknown type name 'uint32_t'
  11:16.33   const uint32_t kBufferSize = 4 * 1024 - 1;
  11:16.34         ^
  11:16.44 2 errors generated.
  (et al)

Differential Revision: https://phabricator.services.mozilla.com/D167491

diff --git a/gfx/2d/Rect.h b/gfx/2d/Rect.h
--- a/gfx/2d/Rect.h
+++ b/gfx/2d/Rect.h
@@ -10,16 +10,17 @@
 #include "BaseRect.h"
 #include "BaseMargin.h"
 #include "NumericTools.h"
 #include "Point.h"
 #include "Tools.h"
 #include "mozilla/Maybe.h"
 
 #include <cmath>
+#include <cstdint>
 
 namespace mozilla {
 
 template <typename>
 struct IsPixel;
 
 namespace gfx {
 
diff --git a/toolkit/components/telemetry/pingsender/pingsender.cpp b/toolkit/components/telemetry/pingsender/pingsender.cpp
--- a/toolkit/components/telemetry/pingsender/pingsender.cpp
+++ b/toolkit/components/telemetry/pingsender/pingsender.cpp
@@ -1,14 +1,15 @@
 /* -*- Mode: C++; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include <cstdlib>
+#include <cstdint>
 #include <cstring>
 #include <ctime>
 #include <fstream>
 #include <iomanip>
 #include <string>
 #include <vector>
 
 #include <zlib.h>


